src/inverted_index.c

```diff
@@ -120,6 +119,14 @@ void IndexReader_OnReopen(void *privdata) {
     }
   }
 
+  IndexReader_OnReopen(ir);
+}
+
+void IndexReader_OnReopen(IndexReader *ir) {
```

Comment:
good splitting out shared code to its own function. removes unnecessary duplication.